File: Techschool/Techschool/ClientApp/src/app/models/vacations/social-with-pregnancy-or-look-vacation.model.ts
import { VacationModel } from './vacation.model';

export class SocialWithPregnancyOrLookVacationModel extends VacationModel {
  typeOfVacation: string;
  orderNumber: string;
  orderDate: Date;

  constructor (data: SocialWithPregnancyOrLookVacationModel = null) {
    super();
    if (data) {
      this.id = data.id;
      this.typeOfVacation = data.typeOfVacation;
      this.startOfVacationDate = new Date(data.startOfVacationDate);
      this.endOfVacationDate = new Date(data.endOfVacationDate);
      this.orderNumber = data.orderNumber;
      this.orderDate = new Date(data.orderDate);
      this.workingYearId = data.workingYearId;
    }
  }
}

File: Techschool/Techschool.BLL/Models/Vacations/SocialWithPregnancyOrLookVacationModel.cs
using System;

namespace Techschool.BLL.Models.Vacations
{
    public class SocialWithPregnancyOrLookVacationModel : VacationModel
    {
        public string TypeOfVacation { get; set; }
        public string OrderNumber { get; set; }
        public DateTime OrderDate { get; set; }
    }
}
